### PR TITLE
[Home Page Picker] Connect Preview Button

### PR DESCRIPTION
**Pull request by chipsnyder**
**Fixes** https://github.com/wordpress-mobile/gutenberg-mobile/issues/2720

**Related PRs**
- `WordPressUI-iOS`: https://github.com/wordpress-mobile/WordPressUI-iOS/pull/76

## To test:
📓 This doesn't include the [PR](https://github.com/wordpress-mobile/WordPress-iOS/pull/15303) to enable HPP by default so you'll still need to toggle the feature on if not already enabled

<details>
<summary><strong>To disable or enable the development version of Home Page Picker</strong></summary>

- Open the app from the build that allows FeatureFlags such as a PR build or a local development build
    - By default, the modal layout picker will be disabled.
    - From the site page:
        - Click on your Gravatar.
        - Click on App Settings.
        - Click on Debug.
        - Toggle "Site Creation: Home Page Picker" to enable or disable the picker
        - <kdb><a href="https://user-images.githubusercontent.com/3384451/97343011-5f032980-185d-11eb-9030-06c17f732fec.gif"></a></kdb>

</details>

When enabled, the first step of the Site Creation flow will be replaced with the Home Page Picker flow.

### Page Renders
1. Navigate to the Create WordPress.com site
    - My Sites > ➕ > Create WordPress.com site
1.  Select a site design
1. Select Preview
1. **Expect** 
    - To see a progress bar move across the screen as the web page loads
    - To see the progress bar disappear once the screen loads
    - To see a web view load with a demo style web page
    - The web view page should be minimally interactive. For example, the menu, if available, will expand when clicked but clicking on buttons will only highlight them

### Can Create Site
1. Navigate to the Create WordPress.com site
    - My Sites > ➕ > Create WordPress.com site
1.  Select a site design
1. Select Preview
1. Select Choose
1. Search for and Select a Domain
1. Click create site
1. **Expect** to have a site created with a home page that matches the previewed design

### Network Glitch
1. Navigate to the Create WordPress.com site
    - My Sites > ➕ > Create WordPress.com site
1. Wait for site designs to load
1. Disengage your network connection
1.  Select a site design
1. Select Preview
1. **Expect** to see a no network warning but allow the "Choose" button
     - I set it up this way in case the demo site is down for some reason or if the user is just on a bad connection. Plus they would be able to hit back and select choose if this did occur. 
1. Repair your network connection
1. Select Choose
1. Search for and Select a Domain
1. Click create site
1. **Expect** to have a site created with a home page that matches the previewed design

### Analytics Successful load
📓 To help with testing these you can filter the console logs on `Tracked:`

1. Navigate to the Create WordPress.com site
    - My Sites > ➕ > Create WordPress.com site
1.  Select a site design
1. Select Preview
    - **Expect** to see the event `enhanced_site_creation_site_design_preview_viewed` fire with the parameter `template` populated with the selected design's `slug`
1. When the URL starts loading
    -  **Expect** to see the event `enhanced_site_creation_site_design_preview_loading`
1. When the URL successfully loads
    -  **Expect** to see the event `enhanced_site_creation_site_design_preview_loaded`
1. Select Choose
    - **Expect** to see the event `enhanced_site_creation_site_design_selected` fire with the parameter `template` 

### Analytics Network Glitch load
📓 To help with testing these you can filter the console logs on `Tracked:`

1. Navigate to the Create WordPress.com site
    - My Sites > ➕ > Create WordPress.com site
1.  Select a site design
1. Disengage your network connection
1. Select Preview
    - **Expect** to see the event `enhanced_site_creation_site_design_preview_viewed` fire with the parameter `template` populated with the selected design's `slug`
1. When the URL starts loading
    -  **Expect** to see the event `enhanced_site_creation_site_design_preview_loading`
1. When the URL fails to load
    -  **Expect** to see the event `enhanced_site_creation_error_shown` fire with the parameter `error_info` with a description of the error
1. Repair your network connection
1. Select Choose
    - **Expect** to see the event `enhanced_site_creation_site_design_selected` fire with the parameter `template` 

## Screenshots:

Preview | No Network
--- | ---
<kdb><a href="https://user-images.githubusercontent.com/3384451/99108312-267d8280-25b5-11eb-8dc8-6f6a157cd476.png"></a></kdb> | <kdb><a href="https://user-images.githubusercontent.com/3384451/99108714-c2a78980-25b5-11eb-9777-3b44e3dc18fa.png"></a></kdb>

## Additional Context: 

This is a continuation of the work done in the [Home Page Picker Project](https://github.com/wordpress-mobile/gutenberg-mobile/labels/Project%3A%20Home%20Page%20Picker) and is part of the [Round 2 requirements](https://github.com/wordpress-mobile/gutenberg-mobile/labels/Project%3A%20Home%20Page%20Picker%20Round%202).  

## PR submission checklist:

- [x] I have considered adding unit tests where possible.
- [x] I have considered adding accessibility improvements for my changes.
- [x] I have considered if this change warrants user-facing release notes and have added them to `RELEASE-NOTES.txt` if necessary.

### CODE REVIEW

**Comment by chipsnyder** [comment]
@kyleaparker @iamthomasbishop Here's a build with the Preview functionality. One thought I had while working on this was wondering if we should include the `Skip` button on preview too or not. WDYT?

**Comment by chipsnyder** [comment]
@kyleaparker @iamthomasbishop Wanted to follow up on [this](https://github.com/wordpress-mobile/WordPress-iOS/pull/15293#issuecomment-728097730). Any thoughts on a `Skip` button for this screen. If you don't have any strong feelings we can leave it off for now and add it later.

**Comment by iamthomasbishop** [comment]
> Any thoughts on a Skip button for this screen. If you don't have any strong feelings we can leave it off for now and add it later.

I don't think I would add a skip button on the preview sheet, necessarily. It's sort of an in-between step, so IMO it would feel a little odd.

This reminds me of something, and forgive me if I missed a decision or some context. Did we decide against going with a modal sheet for this flow? I think I suggested we use the standard close button style `(x)`, but maybe that only makes sense if we go with a modal sheet? 

In other words, if the decision is to stick with the full-screen view, maybe a simple text label of `Cancel` or `Close` would be better. Either way is fine, but we might want to adjust the left-side action in the nav bar to be appropriate for the context.

@kyleaparker or @SylvesterWilmott if you have any thoughts/feelings on the above, happy to hear it ☝️ 😃

**Comment by chipsnyder** [comment]
>I don't think I would add a skip button on the preview sheet, necessarily. It's sort of an in-between step, so IMO it would feel a little odd.

That works for me I'll leave it off. Thanks @iamthomasbishop!

>Did we decide against going with a modal sheet for this flow? 

We didn't make a clear decision on that yet. Carrying over a question from: https://github.com/wordpress-mobile/WordPress-iOS/pull/15231#issuecomment-724965218

> I like that idea, although there might be some issues to consider. That presentation style would force us into supporting landscape mode as well. The choose a domain seems to work ok but not great in landscape mode, and the final confirmation looks a bit funky. I went ahead and set iPad to use the same setting as MLP, which I think looks better. Let me know though what you think about moving forward with iPhones.
> <kdb><a href="https://user-images.githubusercontent.com/3384451/98733255-3fe0bd80-236e-11eb-9969-625841ceb99b.png"></a></kdb>

The confirmation of the site completion looks not super useful but if you don't mind that view I can make that change.

**Review by antonis — APPROVED** [review]
The code LGTM and the app (tested the AppCenter build) behaves as expected 🎉 
Great work Chip 👍

**Comment by chipsnyder** [comment]
RE: https://github.com/wordpress-mobile/WordPress-iOS/pull/15293#issuecomment-730110854 

Summarizing a side conversation with @iamthomasbishop we're going to run forward with the changes to move to a page sheet that I'll tackle in another PR. 

We'll move to a page sheet and:
- Change the (x) (close button) on the initial sheet to a text label of Cancel
- On the preview, we'll keep the (x) button, but move it to the right side — just like on the main MLP

The modal with in a modal flow is good and the site confirmation is good.
